src/client/ua_client.c

```diff
@@ -16,9 +16,11 @@
 
 static void UA_Client_init(UA_Client* client, UA_ClientConfig config) {
     client->state = UA_CLIENTSTATE_READY;
-    UA_Connection_init(&client->connection);
-    UA_SecureChannel_init(&client->channel);
-    client->channel.connection = &client->connection;
+    client->connection = UA_calloc(1, sizeof(UA_Connection));
```

Comment:
*idention
*calloc and init is a double affort, I'd suggest to go for change calloc to malloc